### DISCUSSION TITLE
CLI Error 7: InvalidArgs for networkdiagnostic get TLV 27, 28

### DISCUSSION THREAD

**Discussion by pphrump** [openthread/openthread#9046] [Q&A] [answered]
Cli command, "networkdiagnostic get" errors for TLV(s) 27 and or 28.  Tested on 5/12/2023.

Using the OpenThread Border Router guide (https://openthread.io/guides/border-router/docker/).  
openthread git status shows branch is up to date with origin/main, docker images obtained via: docker pull openthread/otbr:latest on 5/12/2023.

Testing includes the docker border router, simulated ftd and mtd.  Command issued: networkdiagnostic get <rlatn ipv6address> 0 1 2 3 4 5 6 7 8 9 14 15 16 17 18 24 25 26 27 28.    Result: Error 7: InvalidArgs.  When TLVs 27 and 28 are removed, the command executes and the YAML output is displayed.  The output from the test is attached.
[networkdiagnosticOutput05122023.txt](https://github.com/openthread/openthread/files/11465379/networkdiagnosticOutput05122023.txt)

**Comment by jwhui** [comment, marked as answer]
The number of TLV types that can be requested is limited by https://github.com/openthread/openthread/blob/48abf12e00bc36f403ef7c016c771aea196ae40f/include/openthread/netdiag.h#L55

You can try increasing this value.

**Comment by pphrump** [comment]
Thanks for the reply.  Sorry for the oversight on my part.  I had thought once the new TLVs were added, that would have been adjusted as well.
